### ISSUE TITLE
Memory leak using minimal web server?

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#11092]
|<img alt="chileverde's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| chileverde reported|
|-|-|
|Trac ID|trac#1495|
|Type|defect|
|Created|2006-02-20 01:13:11Z|

Attachments:

* [code.tar.gz](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/b95/b957056dd8ad16f1f46ca090727df5828d933758/b3fdfcb267e561ccbc216d7e9bdff1627d00c409.gz) (517 bytes) - added by chileverde on 2006-02-20 01:13:11Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__1495 1495
type__defect defect
reporter__chileverde chileverde
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__web web
keywords__ 
time__1140397991000000 1140397991000000
changetime__1141101829000000 1141101829000000
version__ 
owner__jknight jknight
cc__jknight cc__chileverde cc__itamarst
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="chileverde's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|chileverde commented|
|-|-|

```
#!html
<pre>
It seems there is a mem leak somewhere. When I start the attached server with:       
twistd -y server.py       
is uses around 7mb of mem, but then I run:       
ab -n10000 -c4 http://host:8080/    (from a different pc of course)   
memory starts getting used. When it finishes running, the memory is still       
used up to (96% of 128 mb), and it doesn't free up until I kill the twisted       
process.       
       
System:    
Ubuntu 2.6.10-5-386   
Python 2.4.1 (#2, Mar 30 2005, 21:51:10)   
[GCC 3.3.5 (Debian 1:3.3.5-8ubuntu2)] on linux2   
   
Please let me know if I can help in any way.
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="chileverde's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|chileverde commented|
|-|-|

```
#!html
<pre>
Sorry for the double post. Also forgot to add:   
>>> import twisted  
>>> twisted.__version__  
'2.2.0'  
>>>
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="jyknight's avatar" src="https://avatars.githubusercontent.com/u/873903?s=50" width="50" height="50">](https://github.com/jyknight)|@jyknight commented|
|-|-|

Verified leak; appears to be because of the components backward compatibility layer -- removing components.backwardsCompatImplements(Resource) from resource.py made the leak not appear.

Probably the "_fixedClasses" dictionary in components.py should be a weak hash, itamar?

**Comment by twisted-trac** [comment]
|<img alt="foom's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|foom set status to `closed`|
|-|-|

(In [16085]) Use an attribute on the class to store fixedness instead of a hashtable.
This removes the leak when creating and discarding classes.

Fixes [#11092](https://github.com/twisted/twisted/issues/11092)
